iothub/device/src/Pipeline/TransportHandler.cs

```diff
@@ -18,7 +18,7 @@ protected TransportHandler(PipelineContext context, IotHubClientTransportSetting
 
         public override Task WaitForTransportClosedAsync()
         {
-            _transportShouldRetry = new TaskCompletionSource<bool>();
+            _transportShouldRetry ??= new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
This looks like it may have had a bug. If someone called this multiple times, we'd overwrite the previous TCS, and previous waiters would be left hanging.

But do we need to reset and add a new one sometime? I'm not sure.

@timtay-microsoft and @abhipsaMisra, thoughts here?

Reply:
I don't have a lot of context on this one. It does seem like the CloseAsync() call should end when the connection has been closed, and that would spare this layer from using TCS. But that is a problem for another PR

Reply:
I think things like this need to be reconceived a bit when we tackle open/close/open, too.

Reply:
I agree that this entire class needs to be tackled on its own.